src/js/containers/recipient/topFive/TopFiveContainer.jsx

```diff
@@ -0,0 +1,166 @@
+/**
+ * TopFiveContainer.jsx
+ * Created by Kwadwo Opoku-Debrah 07/10/18
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { isCancel } from 'axios';
+
+import {
+    getTrailingTwelveMonths,
+    convertFYToDateRange,
+    currentFiscalYear,
+    earliestFiscalYear
+} from 'helpers/fiscalYearHelper';
+import * as SearchHelper from 'helpers/searchHelper';
+import BaseStateCategoryResult from 'models/v2/state/BaseStateCategoryResult';
+
+import TopFive from 'components/recipient/topFive/TopFive';
+
+const propTypes = {
+    total: PropTypes.number,
+    category: PropTypes.string,
+    fy: PropTypes.string,
+    recipientHash: PropTypes.string
+};
+
+export class TopFiveContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            loading: true,
+            error: false,
+            results: []
+        };
+
+        this.request = null;
+    }
+
+    componentDidMount() {
+        this.loadCategory();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.recipientHash !== this.props.recipientHash) {
+            this.loadCategory();
+        }
+        if (prevProps.fy !== this.props.fy) {
+            this.loadCategory();
+        }
+    }
+
+    dataParams() {
+        let timePeriod = null;
+        if (this.props.fy === 'latest') {
+            const trailing = getTrailingTwelveMonths();
+            timePeriod = {
+                start_date: trailing[0],
+                end_date: trailing[1]
+            };
+        }
+        else if (this.props.fy === 'all') {
+            const currentFY = currentFiscalYear();
+            const earliestRange = convertFYToDateRange(parseInt(earliestFiscalYear, 10) + 1);
```

Comment:
This should be `(parseInt(earliestFiscalYear, 10))` (Fiscal year 2008 starts Oct 1 2007)